### DISCUSSION TITLE
wallet is undefined when used as provider inside snap

### DISCUSSION THREAD

**Discussion by Arch0125** [MetaMask/snaps#1271] [Q&A] [answered]
I was using the following provider  ``` const provider = new ethers.providers.Web3Provider(wallet as any);```
However, the wallet in undefined hence cannot get the signer as well, couldn't use JSON-RPC providers as I need to make eth_requestAccounts call.
Any workaround ?

**Comment by ziad-saab** [comment, marked as answer]
Since the new Flask release you have to use `ethereum` instead of wallet. You also need to request [permission](https://docs.metamask.io/guide/snaps-permissions.html#endowment-ethereum-provider) to use it.

**Comment by Arch0125** [comment]
Thanks!!
